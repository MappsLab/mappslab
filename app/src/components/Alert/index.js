// @flow

export { AlertProvider, AlertConsumer } from './AlertProvider'
export { default as Alert } from './Alert'
