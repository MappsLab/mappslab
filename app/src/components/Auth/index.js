// @flow
export { ViewerRoute } from './ViewerRoute'
