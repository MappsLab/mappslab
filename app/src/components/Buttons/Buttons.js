// @flow
import React from 'react'
import styled, { css } from 'styled-components'
import { Link } from 'react-router-dom'

export const ButtonWrapper = styled.button`
	${({ theme, level, disabled }) => css`
		display: inline-block;
		cursor: pointer;
		margin: ${level === 'tertiary' ? theme.layout.spacing.quarter : theme.layout.spacing.single} auto;	
		padding: ${level === 'tertiary' ? theme.layout.spacing.half : theme.layout.spacing.single};
		font-size: ${level === 'tertiary' ? theme.font.size.h5 : theme.font.size.p};
		font-weight: ${level === 'primary' ? theme.font.weight.semi : theme.font.weight.normal};
		border-radius: ${theme.mixins.borderRadius};
		box-shadow: ${level === 'tertiary' ? '' : theme.mixins.boxShadow.normal};
		background-color: ${level === 'primary' ? theme.color.primary.normal : 'white'};
		color: ${
			level === 'primary'
				? theme.color.primary.text
				: level === 'secondary '
				? theme.color.primary.normal
				: theme.color.middleGray
		};
		border: ${level === 'secondary' ? `1px solid ${theme.color.primary.normal}` : ''};
		transition: 0.1s;
		opacity: ${disabled ? 0.6 : 1};
		pointer-events: ${disabled ? 'none' : 'auto'};


		&:hover {
			background-color: ${level === 'primary' ? theme.color.primary.accent : ''};
			box-shadow: ${level === 'tertiary' ? theme.mixins.boxShadow.normal : ''};

		`};
	}
`

type ButtonProps = {
	type?: 'button' | 'submit' | 'reset',
	level?: 'primary' | 'secondary' | 'tertiary',
	onClick?: () => any | (() => Promise<any>),
	to?: string,
	as?: string,
}

export const Button = (props: ButtonProps) => (
	<ButtonWrapper {...props} as={props.as || (props.to ? Link : undefined)} type={props.to || props.as ? undefined : props.type} />
)

Button.defaultProps = {
	type: 'button',
	level: 'primary',
	onClick: undefined,
	to: undefined,
	as: undefined,
}

type SubmitButtonProps = {
	disabled: boolean,
}

export const SubmitButton = (props: SubmitButtonProps) => <ButtonWrapper {...props} type="submit" level="primary" />
