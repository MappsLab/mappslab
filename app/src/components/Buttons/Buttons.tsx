import React from 'react'
import styled, { css, DefaultTheme } from 'styled-components'
import { Link } from 'react-router-dom'
import { ToolTip } from '../ToolTip'

interface ButtonWrapperProps {
	theme: DefaultTheme
	disabled: boolean
	level: string
}

export const ButtonWrapper = styled.button`
	${({ theme, level, disabled }: ButtonWrapperProps) => css`
		display: inline-flex;
		align-items: center;
		cursor: pointer;
		margin: ${level === 'tertiary'
				? theme.layout.spacing.quarter
				: theme.layout.spacing.single}
			auto;
		padding: ${level === 'tertiary'
			? theme.layout.spacing.half
			: theme.layout.spacing.single};
		font-size: ${level === 'tertiary' ? theme.font.size.h5 : theme.font.size.p};
		font-weight: ${level === 'primary'
			? theme.font.weight.semi
			: theme.font.weight.normal};
		border-radius: ${theme.mixins.borderRadius};
		box-shadow: ${level === 'tertiary' ? '' : theme.mixins.boxShadow.normal};
		background-color: ${level === 'primary'
			? theme.color.primary.normal
			: 'white'};
		color: ${level === 'primary'
			? theme.color.primary.text
			: level === 'secondary '
			? theme.color.primary.normal
			: theme.color.middleGray};
		border: ${level === 'secondary'
			? css`1px solid ${theme.color.primary.normal}`
			: ''};
		transition: 0.1s;
		${disabled &&
			css`
				opacity: 0.6;
				pointer-events: none;
			`}

		& > svg {
			margin: 0 0.5em;
		}

		&:hover {
			background-color: ${level === 'primary'
				? theme.color.primary.accent
				: ''};
			box-shadow: ${level === 'tertiary' ? theme.mixins.boxShadow.normal : ''};
		}
	`}
`

interface ButtonProps {
	type?: string
	level?: string
	onClick?: () => any | (() => Promise<any>)
	tooltip?: string
	to?: string
	as?: string
	children?: React.ReactNode
	disabled?: boolean
	htmlFor?: string
}

export const Button = (props: ButtonProps) => (
	<ToolTip message={props.tooltip}>
		<ButtonWrapper
			{...props}
			// @ts-ignore
			as={props.as || (props.to ? Link : undefined)}
			type={props.to || props.as ? undefined : props.type}
		/>
	</ToolTip>
)

Button.defaultProps = {
	type: 'button',
	level: 'primary',
	tooltip: undefined,
	onClick: undefined,
	to: undefined,
	as: undefined,
}

interface SubmitButtonProps extends ButtonProps {
	wide: boolean
}

export const SubmitButton = ({ as, ...rest }: SubmitButtonProps) => (
	<ButtonWrapper
		// @ts-ignore
		as={as}
		{...rest}
		type="submit"
		level="primary"
	/>
)
