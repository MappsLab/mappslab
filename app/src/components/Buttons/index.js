export * from './Buttons'
export { default as RoundButton } from './RoundButton'
