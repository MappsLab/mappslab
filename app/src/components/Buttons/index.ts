export * from './Buttons'
export * from './RoundButton'
