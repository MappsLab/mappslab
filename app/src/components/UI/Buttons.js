// @flow
import React from 'react'
import styled from 'styled-components'

export const ButtonWrapper = styled.button`
	${({ theme, level }) => `
		display: block;
		margin: ${level === 'tertiary' ? theme.layout.spacing.quarter : theme.layout.spacing.single} auto;	
		padding: ${level === 'tertiary' ? theme.layout.spacing.half : theme.layout.spacing.single};
		font-size: ${level === 'tertiary' ? theme.text.size.h5 : theme.text.size.p};
		font-weight: ${level === 'primary' ? theme.text.weight.semi : theme.text.weight.normal};
		border-radius: ${theme.mixins.borderRadius};
		box-shadow: ${level === 'tertiary' ? '' : theme.mixins.boxShadow.normal};
		background-color: ${level === 'primary' ? theme.color.primary.normal : 'white'};
		color: ${
			level === 'primary'
				? theme.color.primary.text
				: level === 'secondary '
					? theme.color.primary.normal
					: theme.color.middleGray
		};
		border: ${level === 'secondary' ? `1px solid ${theme.color.primary.normal}` : ''};
		transition: 0.1s;

		&:hover {
			background-color: ${level === 'primary' ? theme.color.primary.accent : ''};
			box-shadow: ${level === 'tertiary' ? theme.mixins.boxShadow.normal : ''};

		`};
	}
`

type ButtonProps = {
	type?: 'button' | 'submit' | 'reset',
	level?: 'primary' | 'secondary' | 'tertiary',
}

export const Button = (props: ButtonProps) => <ButtonWrapper {...props} />

Button.defaultProps = {
	type: 'button',
	level: 'primary',
}

type SubmitButtonProps = {
	disabled: boolean,
}

export const SubmitButton = (props: SubmitButtonProps) => <ButtonWrapper {...props} type="submit" level="primary" />
