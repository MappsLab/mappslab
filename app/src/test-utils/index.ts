export * from '@testing-library/react'
export * from './mockServer'
// @ts-ignore
export * from './render'
