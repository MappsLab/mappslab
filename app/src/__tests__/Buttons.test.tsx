/* eslint-disable no-undef */
import React from 'react'
import { render } from '../test-utils'
import { Button } from '../components/Buttons'
import { StaticRouter } from 'react-router-dom'

const buttonText = 'Click Me!'

describe('Button Component', () => {
	it('Mounts Correctly', () => {
		const { container } = render(<Button>{buttonText}</Button>)
		expect(container).toMatchSnapshot()
	})

	it('Renders as a <Link> if given a `to` prop', () => {
		const { container } = render(
			<StaticRouter location="/" context={{}}>
				<Button to="/path">Go</Button>
			</StaticRouter>,
		)
		expect(container.getElementsByTagName('a')).toHaveLength(1)
	})
})
