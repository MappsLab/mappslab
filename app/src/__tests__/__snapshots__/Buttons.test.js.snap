// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Mounts Correctly 1`] = `
.c0 {
  display: inline-block;
  cursor: pointer;
  margin: 12px auto;
  padding: 12px;
  font-size: 16px;
  font-weight: 500;
  border-radius: 2px;
  box-shadow: 0 1px 2px rgba(0,0,0,0.5);
  background-color: #3930dc;
  color: white;
  -webkit-transition: 0.1s;
  transition: 0.1s;
}

.c0:hover {
  background-color: #3E34FF;
}

<div>
  <button
    class="c0"
    type="button"
  >
    Click Me!
  </button>
</div>
`;
